File: src/java/WebNinja/src/test/java/webninja/workflow/ProjectWorkflow.java
package webninja.workflow;

import webninja.AssignedIssuesReport;
import webninja.Issue;
import webninja.selenuim.*;

import java.util.List;

public class ProjectWorkflow {
    private final CodeTrack codeTrack;
    private String projectName;

    public ProjectWorkflow(CodeTrack codeTrack) {

        this.codeTrack = codeTrack;
    }

    public ProjectWorkflow addIssues(List<Issue> issues) {
        for (Issue issue : issues) {
            IssueForm issueForm = codeTrack.showNewIssueForm();
            issueForm.setProjectName(projectName);
            issueForm.setTitle(issue.getTitle());
            issueForm.setDescription(issue.getTitle());
            issueForm.setSeverity(issue.getSeverity());
            codeTrack.submit();
        }
        return this;
    }

    public String getProjectName() {
        return projectName;
    }

    public ProjectWorkflow assignIssueToUser(String issueTitle, String user) {
        HomePage homePage = codeTrack.gotoHomePage();
        codeTrack.setProjectName(projectName);
        IssuePage issuePage = homePage.showIssueWithTitle(issueTitle);
        IssueForm issueForm = issuePage.startEdit();
        issueForm.assignTo(user);
        codeTrack.submit();
        return this;
    }

    public AssignedIssuesReport viewAssignedIssuesReport() {
        ReportsPage reportsPage = codeTrack.gotoReportsPage();
        IssueTable issueTable = reportsPage.showIssuesAssignedToLoggedInUser();
        return new AssignedIssuesReport(issueTable.getIssues());
    }

    public void setCurrentProject(String project) {
        this.projectName = project;
        codeTrack.setProjectName(projectName);
    }
}

File: src/dotnet/WebNinja/WatiNinja/watininja/workflow/ProjectWorkflow.cs
using System;
using System.Collections.Generic;
using WatiNinja.watininja.technical;

namespace WatiNinja.watininja.workflow
{
    public class ProjectWorkflow
    {
        private readonly CodeTrack _codeTrack;
        public ProjectWorkflow(CodeTrack codeTrack)
        {
            _codeTrack = codeTrack;
        }

        public string CurrentProject { set; get; }

        public ProjectWorkflow AddIssues(IList<Issue> issues)
        {
            foreach (var issue in issues)
            {
                IssueForm issueForm = _codeTrack.GotoNewIssueForm();
                issueForm.ProjectName = CurrentProject;
                issueForm.Title = issue.Title;
                issueForm.Description = issue.Title;
                issueForm.Severity = issue.Severity;
                issueForm.Submit();

            }
            return this;
        }

        public ProjectWorkflow AssignIssueToUser(string issueTitle, String user)
        {
            HomePage homePage = _codeTrack.GotoHomePage();
            homePage.ProjectName = CurrentProject;
            IssuePage issuePage = homePage.ShowIssueWithTitle(issueTitle);
            IssueForm issueForm = issuePage.StartEdit();
            issueForm.AssignTo(user);
            issueForm.Submit();
            return this;
        }

        public AssignedIssuesReport ViewAssignedIssuesReport()
        {
            ReportsPage reportsPage = _codeTrack.GotoReportsPage();
            IssueTable issueTable = reportsPage.ShowIssuesAssignedToLoggedInUser();
            return new AssignedIssuesReport(issueTable.Issues);
        }
    }
}
